src/rdf4cpp/rdf/storage/node/NodeStorage.hpp

```diff
@@ -17,88 +17,332 @@ namespace rdf4cpp::rdf::storage::node {
 
 class INodeStorageBackend;
 
+/**
+ * NodeStorage provides an interface to the internal storage <div>Node</div>s.
+ * Each NodeStorage has a INodeStorageBackend, which is uniquely identified by a identifier::NodeStorageID.
+ * There can be at most 1024 different INodeStorageBackend instances at once.
+ * If no NodeStorage with the identifier::NodeStorageID of a certain INodeStorageBackend exists anymore, the corresponding INodeStorageBackend is destructed.
+ * This does not apply to the default_instance.
+ * The live cycle of NodeStorage's and their backends is managed by the static methods within this class.
+ */
 class NodeStorage {
```

Comment:
@nkaralis I tried to address the points from your review https://github.com/rdf4cpp/rdf4cpp/pull/21#issuecomment-1025506011 . Could you check if you think that this reworked interface is an improvement and reevaluate?

Reply:
The split of the functions looks good to me.